proofs/tacmach.mli

```diff
@@ -109,6 +109,7 @@ module New : sig
   val pf_concl : ([ `NF ], 'r) Proofview.Goal.t -> types
 
   val pf_unsafe_type_of : ('a, 'r) Proofview.Goal.t -> Term.constr -> Term.types
```

Comment:
Can you add a comment describing what `pf_unsafe_type_of` and explain the pb here? It will prevent other similar issues, and will remind us to inspect remaining calls to `pf_unsafe_type_of`.